### ISSUE TITLE
only functional on a main monitor in a dual monitor setup

### ISSUE THREAD

**Issue by khraibani-h** [stephane-13/gnome-shell-extension-dim-background-windows#16]
i would like to ask if this is normal behavior to be only functional on a main monitor in a dual monitor setup.
so even if on the extended monitor their is a full screen window open it will be greyed out.

**Comment by stephane-13** [comment]
You should be able to control if the dimming effect is enabled on the primary monitor, on the secondary monitor(s) or both in the preferences ("Apply the dimming effect to:").

There is another option, which controls the dimming effect for maximized / full screen windows ("Apply the dimming effect to maximized windows:").

Let me know if those options don't work as expected with your setup and I'll have a look.

**Comment by joser93** [comment]
Greetings

I am testing this on Gnome 46, recently installed the extension, but I found out that the Apply the dimming effect to the pramary monitor only, also affects my second monitor on extend configuration, and the second monitor only, does not apply any effect on both monitors. Maybe should be reviewed in a certain way? 

If any information is needed to fix the issue, let me know.

**Comment by stephane-13** [comment]
Would you be able to manually install version 16 - not yet published - from the link below?
- https://github.com/stephane-13/gnome-shell-extension-dim-background-windows/blob/v16_gnome45-46/dim-background-windows%40stephane-13.github.com.shell-extension.zip

This should fix the secondary monitor behavior, hopefully.

**Comment by piobra** [comment]
@stephane-13 I tried with version 16 but it gives an error that gschemas.compiled file is not found

**Comment by stephane-13** [comment]
The file gschemas.compiled is created automatically when installing the extension with the gnome-extensions tool, like this:

`gnome-extensions install dim-background-windows@stephane-13.github.com.shell-extension.zip --force`

You should then find the file as `~/.local/share/gnome-shell/extensions/dim-background-windows\@stephane-13.github.com/schemas/gschemas.compiled`

If you installed the extension manually - by unzipping the file - you won't get the gschemas.compiled file.
It can then be created by running:

`
glib-compile-schemas ~/.local/share/gnome-shell/extensions/dim-background-windows\@stephane-13.github.com/schemas 
`

But the recommended way is to use the gnome-extensions tool for the installation.

**Comment by piobra** [comment]
@stephane-13 
thanks, it doesn't seem to have fixed the issue. It only dims the window on the secondary monitor when the window focused on primary is not maximized. When all windows are maximized it still only dims the primary monitor window but not vice versa. I enabled all dimming settings.

**Comment by stephane-13** [comment]
I finally had time to hook up a secondary monitor on my system.
I confirm that a maximized window on the primary monitor isn't dimmed when a window on another monitor has the focus.

However nothing in the extension is causing this; it seems to be a bug within Gnome.
It's possible to see that weird things are going on by:

1) Maximizing a window on the primary monitor
2) Clicking on a window on the secondary monitor
3) Right-clicking on the desktop wallpaper of the secondary monitor several times and/or hovering the mouse on the Gnome taskbar on any monitor

What happens is that the maximized window is getting / loosing the dimming effect each time the popup menu is open / closed, and randomly when hovering the mouse on the taskbar. This happens without the extension doing anything.

The bottom line is: I can't fix this weird behavior in the extension.
The shader is correctly applied to the windows - including the maximized one on the primary display - but for some reason Gnome decides to not use it unless weird things are done, like opening that context menu or hovering the mouse on the Gnome taskbar.

I'm leaving this issue open and I'll monitor future Gnome/Mutter versions to see if either it's fixed or if I can implement some workaround.
